BUG/TST: stats: New XSLOW failures in stats

Running the XSLOW tests produces two new failures in `stats` that I haven't seen before.  These appear to be the result of https://github.com/scipy/scipy/pull/23739.  I see the failures in both Linux and Mac.

This command runs just the relevant test class:

```
$ SCIPY_XSLOW=1 spin test -m full -v -s stats -- -k TestMGCStat
Invoking `build` prior to running tests:
$ meson compile -j 12 -C build
INFO: autodetecting backend as ninja
INFO: calculating backend command to run: /home/warren/py3.13.7/bin/ninja -C /home/warren/repos/git/forks/scipy-fresh/build -j 12
ninja: Entering directory `/home/warren/repos/git/forks/scipy-fresh/build'
[2/338] Generating subprojects/highs/src/HConfig.h with a custom command
$ meson install --only-changed -C build --destdir ../build-install --tags=runtime,python-runtime,tests,devel
$ export PYTHONPATH="/home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.13/site-packages"
$ /home/warren/py3.13.7/bin/python3.13 -P -c 'import scipy'
$ cd /home/warren/repos/git/forks/scipy-fresh/build-install/usr/lib/python3.13/site-packages
$ /home/warren/py3.13.7/bin/python3.13 -P -m pytest -v -k TestMGCStat --pyargs scipy.stats
======================================= test session starts =======================================
platform linux -- Python 3.13.7, pytest-8.4.2, pluggy-1.6.0 -- /home/warren/py3.13.7/bin/python3.13
cachedir: .pytest_cache
hypothesis profile 'default'
rootdir: /home/warren/repos/git/forks/scipy-fresh
configfile: pytest.ini
plugins: xdist-3.8.0, anyio-4.11.0, timeout-2.4.0, hypothesis-6.138.14, cov-6.2.1
collected 16927 items / 16916 deselected / 1 skipped / 11 selected                                

scipy/stats/tests/test_mgc.py::TestMGCStat::test_oned[linear-0.97-0.001] PASSED             [  9%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_oned[nonlinear-0.163-0.001] PASSED         [ 18%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_oned[independence--0.0012-0.3646] FAILED   [ 27%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_fived[linear-0.184-0.001] PASSED           [ 36%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_fived[nonlinear-0.113-0.001] FAILED        [ 45%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_twosamp PASSED                             [ 54%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_workers PASSED                             [ 63%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_random_state PASSED                        [ 72%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_dist_perm PASSED                           [ 81%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_pvalue_literature PASSED                   [ 90%]
scipy/stats/tests/test_mgc.py::TestMGCStat::test_alias PASSED                               [100%]

============================================ FAILURES =============================================
_______________________ TestMGCStat.test_oned[independence--0.0012-0.3646] ________________________

self = <scipy.stats.tests.test_mgc.TestMGCStat object at 0x7e89d1fd7e10>, sim_type = 'independence'
obs_stat = -0.0012, obs_pvalue = 0.3646

    @pytest.mark.xslow
    @pytest.mark.parametrize("sim_type, obs_stat, obs_pvalue", [
        # Reference values produced by `multiscale_graphcorr` ->
        # this only guards against unintended changes
        ("linear", 0.97, 1/1000),           # test linear simulation
        ("nonlinear", 0.163, 1/1000),       # test spiral simulation
        ("independence", -0.0012, 0.3646)   # test independence simulation
    ])
    def test_oned(self, sim_type, obs_stat, obs_pvalue):
        # generate x and y
        rng = np.random.default_rng(8157117705)
        x, y = self._simulations(samps=100, dims=1, sim_type=sim_type, rng=rng)
    
        # test stat and pvalue
        stat, pvalue, _ = stats.multiscale_graphcorr(x, y, random_state=rng)
>       assert_approx_equal(stat, obs_stat, significant=1)
E       AssertionError: 
E       Items are not equal to 1 significant digits:
E        ACTUAL: -0.003965353120480147
E        DESIRED: -0.0012

_          = {'mgc_map': array([[ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
      ... -8.90097762e-03,
        9.10596486e-02, -3.46568627e-03,  1.40948704e-03,  1.03201324e-02]), 'opt_scale': [100, 100]}
obs_pvalue = 0.3646
obs_stat   = -0.0012
pvalue     = np.float64(0.4675324675324675)
rng        = Generator(PCG64) at 0x7E89CF2390E0
self       = <scipy.stats.tests.test_mgc.TestMGCStat object at 0x7e89d1fd7e10>
sim_type   = 'independence'
stat       = np.float64(-0.003965353120480147)
x          = array([[-1.52068654],
       [-0.69230394],
       [ 1.09392702],
       [ 0.96150942],
       [ 0.84660868],
       [...971789],
       [ 0.86248042],
       [-0.79664417],
       [-1.34193391],
       [-1.01794253],
       [ 0.61520384]])
y          = array([[ 1.23659883],
       [ 1.19766874],
       [-0.29364227],
       [-0.27363347],
       [ 1.31986471],
       [...556772],
       [ 1.27762522],
       [ 0.80234965],
       [ 1.16541459],
       [ 1.17709771],
       [-0.88476903]])

scipy/stats/tests/test_mgc.py:126: AssertionError
__________________________ TestMGCStat.test_fived[nonlinear-0.113-0.001] __________________________

self = <scipy.stats.tests.test_mgc.TestMGCStat object at 0x7e89d18930b0>, sim_type = 'nonlinear'
obs_stat = 0.113, obs_pvalue = 0.001

    @pytest.mark.xslow
    @pytest.mark.parametrize("sim_type, obs_stat, obs_pvalue", [
        # Reference values produced by `multiscale_graphcorr` ->
        # this only guards against unintended changes
        ("linear", 0.184, 1/1000),           # test linear simulation
        ("nonlinear", 0.113, 0.001),         # test spiral simulation
    ])
    def test_fived(self, sim_type, obs_stat, obs_pvalue):
        # generate x and y
        rng = np.random.default_rng(8157117705)
        x, y = self._simulations(samps=100, dims=5, sim_type=sim_type, rng=rng)
    
        # test stat and pvalue
        stat, pvalue, _ = stats.multiscale_graphcorr(x, y, random_state=rng)
>       assert_approx_equal(stat, obs_stat, significant=1)
E       AssertionError: 
E       Items are not equal to 1 significant digits:
E        ACTUAL: 0.010042844938985069
E        DESIRED: 0.113

_          = {'mgc_map': array([[ 0.        ,  0.        ,  0.        , ...,  0.        ,
         0.        ,  0.        ],
      ...  3.79364506e-04,
       -2.15923597e-03,  1.98158704e-02,  9.75335991e-03, -3.43751456e-03]), 'opt_scale': [100, 100]}
obs_pvalue = 0.001
obs_stat   = 0.113
pvalue     = np.float64(0.21678321678321677)
rng        = Generator(PCG64) at 0x7E89CF196B20
self       = <scipy.stats.tests.test_mgc.TestMGCStat object at 0x7e89d18930b0>
sim_type   = 'nonlinear'
stat       = np.float64(0.010042844938985069)
x          = array([[ 1.26821481e-01,  5.31293143e-01,  3.45692914e-01,
        -7.62513270e-01, -6.00249704e-01],
       [-9.40862...9.66879023e-01],
       [ 6.74016257e-02,  3.08182739e-01,  1.77386998e-01,
         2.82938815e-01,  4.75881226e-01]])
y          = array([[ 2.87683157],
       [-0.53760329],
       [ 1.37366701],
       [-0.9155763 ],
       [ 0.69461036],
       [...734813],
       [ 0.02383658],
       [-0.68001841],
       [ 2.37787941],
       [ 0.51609984],
       [ 0.33521705]])

scipy/stats/tests/test_mgc.py:143: AssertionError
===================================== short test summary info =====================================
FAILED scipy/stats/tests/test_mgc.py::TestMGCStat::test_oned[independence--0.0012-0.3646] - AssertionError: 
FAILED scipy/stats/tests/test_mgc.py::TestMGCStat::test_fived[nonlinear-0.113-0.001] - AssertionError: 
==================== 2 failed, 9 passed, 1 skipped, 16916 deselected in 46.86s ====================

```

<details>
<summary>System details</summary>

```
In [1]: import sys, scipy, numpy; print(scipy.__version__, numpy.__version__, sys.version_info); sc
      ⋮ ipy.show_config()
1.17.0.dev0+git20251023.4490e4f 2.4.0.dev0+git20250926.087615c sys.version_info(major=3, minor=13, micro=7, releaselevel='final', serial=0)
Build Dependencies:
  blas:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  lapack:
    detection method: pkgconfig
    found: true
    include directory: /usr/include/x86_64-linux-gnu/openblas-pthread/
    lib directory: /usr/lib/x86_64-linux-gnu/openblas-pthread/
    name: openblas
    openblas configuration: USE_64BITINT= DYNAMIC_ARCH=1 DYNAMIC_OLDER=1 NO_CBLAS=
      NO_LAPACK= NO_LAPACKE=1 NO_AFFINITY=1 USE_OPENMP=0 generic MAX_THREADS=64
    pc file directory: /usr/lib/x86_64-linux-gnu/pkgconfig
    version: 0.3.20
  pybind11:
    detection method: config-tool
    include directory: unknown
    name: pybind11
    version: 3.0.1
Compilers:
  c:
    commands: cc
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  c++:
    commands: c++
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  cython:
    commands: cython
    linker: cython
    name: cython
    version: 3.1.3
  fortran:
    commands: gfortran
    linker: ld.bfd
    name: gcc
    version: 11.4.0
  pythran:
    include directory: ../../../../../py3.13.7/lib/python3.13/site-packages/pythran
    version: 0.18.0
Machine Information:
  build:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
  cross-compiled: false
  host:
    cpu: x86_64
    endian: little
    family: x86_64
    system: linux
Python Information:
  path: /home/warren/py3.13.7/bin/python3.13
  version: '3.13'
```
</details>

--- a/scipy/stats/tests/test_mgc.py
+++ b/scipy/stats/tests/test_mgc.py
@@ -112,9 +112,9 @@ class TestMGCStat:
     @pytest.mark.parametrize("sim_type, obs_stat, obs_pvalue", [
         # Reference values produced by `multiscale_graphcorr` ->
         # this only guards against unintended changes
-        ("linear", 0.97, 1/1000),           # test linear simulation
-        ("nonlinear", 0.163, 1/1000),       # test spiral simulation
-        ("independence", -0.0012, 0.3646)   # test independence simulation
+        ("linear", 0.970137188683, 0.000999000999),      # test linear simulation
+        ("nonlinear", 0.149843048137, 0.000999000999),   # test spiral simulation
+        ("independence", -0.003965353120, 0.4675324675)  # test independence simulation
     ])
     def test_oned(self, sim_type, obs_stat, obs_pvalue):
         # generate x and y
@@ -123,15 +123,15 @@ class TestMGCStat:
 
         # test stat and pvalue
         stat, pvalue, _ = stats.multiscale_graphcorr(x, y, random_state=rng)
-        assert_approx_equal(stat, obs_stat, significant=1)
-        assert_approx_equal(pvalue, obs_pvalue, significant=1)
+        assert_allclose(stat, obs_stat)
+        assert_allclose(pvalue, obs_pvalue)
 
     @pytest.mark.xslow
     @pytest.mark.parametrize("sim_type, obs_stat, obs_pvalue", [
         # Reference values produced by `multiscale_graphcorr` ->
         # this only guards against unintended changes
-        ("linear", 0.184, 1/1000),           # test linear simulation
-        ("nonlinear", 0.113, 0.001),         # test spiral simulation
+        ("linear", 0.193063672972, 0.000999000999),     # test linear simulation
+        ("nonlinear", 0.010042844939, 0.216783216783),  # test spiral simulation
     ])
     def test_fived(self, sim_type, obs_stat, obs_pvalue):
         # generate x and y
@@ -140,8 +140,8 @@ class TestMGCStat:
 
         # test stat and pvalue
         stat, pvalue, _ = stats.multiscale_graphcorr(x, y, random_state=rng)
-        assert_approx_equal(stat, obs_stat, significant=1)
-        assert_approx_equal(pvalue, obs_pvalue, significant=1)
+        assert_allclose(stat, obs_stat)
+        assert_allclose(pvalue, obs_pvalue)
 
     @pytest.mark.xslow
     def test_twosamp(self):

